### DISCUSSION TITLE
How to avoid escaping for an already escaped string in .dump()

### DISCUSSION THREAD

**Discussion by BrandonLiang** [nlohmann/json#2871] [Q&A] [answered]
As titled.

i have the following mapping in the json object:
obj[“text”] = “\”some text\nanother text\””

now, if i do obj.dump(), it will double escape -> “\\\\\”some text\\\nanother text\\\\\”” due to the default escaping.

i wonder if anyone else has run into this issue and how you resolve it. Thanks!

(As a side not, i would like to see an update to include an optional bool param to turn off the default escape option for .dump().)

**Comment by nlohmann** [comment, marked as answer]
There is no way to switch off this escaping as it is required by JSON.

**Reply by nlohmann** [reply]
In more detail: The string `"some text\nanother text"` is not valid JSON, but `\"some text\\nanother text\"` is.

**Reply by kenjinote** [reply]
Is it okay to recognize that the function to restore the escaped string does not exist in this library?

**Reply by nlohmann** [reply]
The lirbary correctly roundtrips any valid UTF-8 string. Not sure what you are looking for.

**Reply by kenjinote** [reply]
What I want to do is assume that I already have a Json encoded UTF-8 string, and I want to decode it and store it in a json object as shown below.Doesn't json_decode function like the one below?

std::string encoded_html = "\\\\u003Chtml>\\\\u003Chead>\\\\u003C/head>\\\\u003Cbody>\\\\u003C/body>\\\\u003C/html>";
json j;
j["html"] = json_decode(encoded_html);

**Reply by nlohmann** [reply]
There is no such function. Use parse and dump.

**Reply by kenjinote** [reply]
I understood. Thank you.
